src/corelibs/U2Gui/src/util/InputWidgetsControllers.cpp

```diff
@@ -239,13 +239,29 @@ void ComboBoxController::addParameterToCmdLineSettings(QStringList &settings) {
     if (cmdLinePrefix.isEmpty()) {
         return;
     }
-    settings << cmdLinePrefix;
+
     int curIndex = inputWidget->currentIndex();
```

Comment:
Can we make it different: always have currentIndex() in the inputWidget valid?
Otherwise we have the following: 1) user see no value selected in the inputWidget, 2) we silently apply some 'default'.

Reply:
I can add a check [here](https://github.com/ugeneunipro/ugene/blob/5e921c2f7f8016dcbd724e9cc191c3a30d7f8464/src/corelibs/U2Gui/src/util/InputWidgetsControllers.cpp#L225) that if the current index has not been set, use the default index (0, for example). What if the combobox is empty?

> silently 

Not silently.

Reply:
> I can add a check [here](https://github.com/ugeneunipro/ugene/blob/5e921c2f7f8016dcbd724e9cc191c3a30d7f8464/src/corelibs/U2Gui/src/util/InputWidgetsControllers.cpp#L225) that if the current index has not been set.

Yes, this looks much better: ComboBox is a UI component and user will see the selected option

> use the default index (0, for example).
0 is good as the last chance.  I would check the 'defaultValue' passed in the constructor first.

> What if the combobox is empty?
The combobox should never be empty or it should be disabled/hidden. Or have an empty value as a valid option.
There should be a SAFE_POINT in the ComboBoxController constructor that options list is not empty.
I understand that this may require to create/fix extra issue or you can add the fix right into this PR: whatever is better for you.

> Not silently.
Dumping message to the log is not an excuse for us when we can avoid this situation completely.